### PR TITLE
Added exploring-solace-native-apis codelab

### PR DESCRIPTION
**Pull request by TomFairbairnSolace**
Added the exploring-solace-native-apis codelab, according to instructions in "Codelab to create a codelab"

### CODE REVIEW

File: markdown/exploring-solace-native-apis/exploring-solace-native-apis.md [hunk 1]
```diff
@@ -0,0 +1,601 @@
+author: Tom Fairbairn
+summary: An exploration of the basics of getting started with Solace's Native APIs
+id: exploring-solace-native-apis
+tags: workshop
+categories: Java
+environments: Web
+status: Draft
+feedback link: A link where users can go to provide feedback (e.g. the git repo or issue page)
```
Comment (TamimiGitHub) [inline markdown/exploring-solace-native-apis/exploring-solace-native-apis.md:8]: Add a feedback link (https://github.com/SolaceSamples for example) or keep it empty here

File: markdown/exploring-solace-native-apis/exploring-solace-native-apis.md [hunk 2]
```diff
@@ -0,0 +1,601 @@
...
+
+![](./img/part-c.png)
+
+### Java: Message Producers
+
+Let's start with Java.  Just like receiving messages, where we have a *Message Consumer*, we need to create a **Message Producer**.  Looking at line 69 of **TopicPublisher.java**, you can see how this object is created:
+
+	XMLMessageProducer prod = session.getMessageProducer(
+
+As for receiving messages, we can provide some callbacks - what happens if there's a response (for request/reply messages), or some kind of error.  In TopicPUblisher we provide an anonymous class to do this.  You could, of course, implement **JCSMPStreamingPublishEventHandler** and pass your class in.
```
Comment (TamimiGitHub) [inline markdown/exploring-solace-native-apis/exploring-solace-native-apis.md:462]: while you're making the edits, minro typo `TopicPUblisher`

File: markdown/exploring-solace-native-apis/exploring-solace-native-apis.md [hunk 3]
```diff
@@ -0,0 +1,601 @@
...
+###6 Publish/Subscribe
+
+**LEAVE SDKPERF RUNNING** and run TopicSubscriber.  The re-run topic publisher.  What you should see is that same message appearing in both applications.  By the magic of Publish/Subscribe, we've sent a single message to two different applications *without either of the original sender or receiver being aware of it*.
+
+![](./img/part-c-2.png)
+
+You could even download and run different TopicSubscribers from different APIs, or different SolaceSamples.  That shows how you can send a message in one language and receive it in different languages.
+
+## Wrap up, and Things We Didn't Cover
+Duration 0:03:00
```
Comment (TamimiGitHub) [inline markdown/exploring-solace-native-apis/exploring-solace-native-apis.md:572]: you forgot a `:` 
`Duration: 0:03:00`
Screen Shot 2020-06-22 at 5 08 01 PM

**Comment by TamimiGitHub** [comment]
LGTM - minor comments

**Comment by TamimiGitHub** [comment]
Here is your preview link https://deploy-preview-51--focused-beaver-3cc79d.netlify.app/

File: markdown/exploring-solace-native-apis/exploring-solace-native-apis.md [hunk 4]
```diff
@@ -0,0 +1,601 @@
...
+summary: An exploration of the basics of getting started with Solace's Native APIs
+id: exploring-solace-native-apis
+tags: workshop
+categories: Java
+environments: Web
+status: Draft
+feedback link: A link where users can go to provide feedback (e.g. the git repo or issue page)
+analytics account: UA-3921398-10
+
+# Exploring Solace Native APIs
```
Comment (TamimiGitHub) [inline markdown/exploring-solace-native-apis/exploring-solace-native-apis.md:11]: I would update the title to say 
`Exploring Solace Native APIs: C & Java`
what do you think?

File: markdown/exploring-solace-native-apis/exploring-solace-native-apis.md [hunk 5]
```diff
@@ -148,31 +148,22 @@ Pick "Solace Messaging" and make a note of the username, password, Message VPN a
...
-	> VM Name: Java HotSpot(TM) 64-Bit Server VM
-	Run Info: Using Java Nanosecond Timer for Timing
-	> Timing Package Clock Speed (Hz): 1000000000
-	> Getting ready to init clients
-	> Adding subscriptions if required
-	> Getting ready to start clients.
-	PUB MR(5s)=    0, SUB MR(5s)=    0, CPU=0
-	PUB MR(5s)=    0, SUB MR(5s)=    0, CPU=0
+Positive
+:  If you're using Docker, you may not connect.  This is because the default docker compose yaml doesn't expose the port Try-Me uses.  Simply stop the containerand edit the .yml file.  Fine the Port Mappings section, and uncomment the line below "#Web transport"
```
Comment (TamimiGitHub) [inline markdown/exploring-solace-native-apis/exploring-solace-native-apis.md:158]: Typo? 
_Find_ the Port...

**Comment by TamimiGitHub** [comment]
Also - looks like for your second commit you forgot to export your markdown file. 
`claat export -ga UA-3921398-10 -o ../../codelabs/ exploring-solace-native-apis.md` to make sure your changes are served

File: markdown/exploring-solace-native-apis/exploring-solace-native-apis.md (hunk 5 above)
Comment (TomFairbairnSolace) [reply markdown/exploring-solace-native-apis/exploring-solace-native-apis.md:158]: I see what you mean, but no, in this instance _expose_ is the right phrase.  You tell Docker which ports on your container you want people to be able to access - you expose the port.

**Pull request merged**